examples/network/c/src/networked-example.c

```diff
@@ -0,0 +1,123 @@
+#include <math.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include "../../../../src/scripting/src/generated/websg.h"
+#include "../../../../src/scripting/src/network.h"
+
+int PACKET_BYTES = 3;
+
+export void *websg_allocate(int size) {
+  return malloc(size);
+}
+
+export void websg_deallocate(void *ptr) {
+  free(ptr);
+}
+
+Node *material_button;
+Node *room1_switch;
+Node *room2_switch;
+Node *left_cube;
+Node *right_cube;
+Node *room1_light;
+Node *room2_light;
+
+Texture *bricks_texture;
+Texture *planks_texture;
+
+bool material_button_state = true;
+bool room1_switch_state = true;
+bool room2_switch_state = true;
+
+unsigned char *outbound_network_packet;
+unsigned char *inbound_network_packet;
+
+export void websg_loaded() {
+  material_button = websg_get_resource_by_name(ResourceType_Node, "MaterialButton");
+  left_cube = websg_get_resource_by_name(ResourceType_Node, "LeftCube");
+  right_cube = websg_get_resource_by_name(ResourceType_Node, "RightCube");
+  room1_switch = websg_get_resource_by_name(ResourceType_Node, "Room1Switch");
+  room2_switch = websg_get_resource_by_name(ResourceType_Node, "Room2Switch");
+  room1_light = websg_get_resource_by_name(ResourceType_Node, "Room1Light");
+  room2_light = websg_get_resource_by_name(ResourceType_Node, "Room2Light");
+
+  MeshPrimitive *right_cube_primitive = right_cube->mesh->primitives[0];
+  bricks_texture = right_cube_primitive->material->base_color_texture;
+  MeshPrimitive *left_cube_primitive = left_cube->mesh->primitives[0];
+  planks_texture = left_cube_primitive->material->base_color_texture;
+
+  Interactable *material_button_interactable = malloc(sizeof(Interactable));
+  Interactable *room1_switch_interactable = malloc(sizeof(Interactable));
+  Interactable *room2_switch_interactable = malloc(sizeof(Interactable));
+
+  websg_create_resource(ResourceType_Interactable, material_button_interactable);
+  websg_create_resource(ResourceType_Interactable, room1_switch_interactable);
+  websg_create_resource(ResourceType_Interactable, room2_switch_interactable);
+
+  material_button->interactable = material_button_interactable;
+  room1_switch->interactable = room1_switch_interactable;
+  room2_switch->interactable = room2_switch_interactable;
+
+  inbound_network_packet = malloc(sizeof(unsigned char[PACKET_BYTES]));
+  outbound_network_packet = malloc(sizeof(unsigned char[PACKET_BYTES]));
+}
+
+export void websg_update(float_t dt) {
+
+  // consume net packets
+  while (websg_network_receive(inbound_network_packet, PACKET_BYTES) == 1) {
```

Comment:
If you make that change where `websg_network_receive` returns the bytes received. Change this to check for `> 0` instead of `== 1`